### PR TITLE
Use URL.extend_query to add params in `ClientRequest`

### PR DESCRIPTION
**Pull request by bdraco**
- [x] needs yarl bumped before this can merge so we test both paths https://github.com/aio-libs/aiohttp/pull/9072

## What do these changes do?

Use `URL.extend_query` to add `params` in `ClientRequest`

## Are there changes in behavior for the user?

no

## Is it a substantial burden for the maintainers to support this?
no

### CODE REVIEW

File: aiohttp/client_reqrep.py [hunk 1]
```diff
@@ -90,6 +90,7 @@
 
 
 _CONTAINS_CONTROL_CHAR_RE = re.compile(r"[^-!#$%&'*+.^_`|~0-9a-zA-Z]")
+_YARL_SUPPORTS_EXTEND_QUERY = tuple(map(int, yarl_version.split(".")[:2])) > (1, 10)
```
Comment (bdraco) [inline aiohttp/client_reqrep.py]:
```suggestion
_YARL_SUPPORTS_EXTEND_QUERY = tuple(map(int, yarl_version.split(".")[:2])) >= (1, 11)
```

**Review by Dreamsorcerer — APPROVED** [review]
I'd be tempted to just move this to 3.11 and increase the minimum yarl version (or do this in 3.10 and remove the old path in 3.11), as these kind of checks tend to be forgotten about for years after they're useful. We can do the same for using the Query type.

**Comment by bdraco** [comment]
I'll backport this to 3.10 and 3.11 and do another PR to increment the yarl version for master/3.11 and cleanup the back compat

**Comment by bdraco** [comment]
probably should merge https://github.com/aio-libs/aiohttp/pull/8564 first and than strip it out of 3.11 and master in another step once yarl requirement is bumped

**Comment by bdraco** [comment]
Pretty significant drop on the profile 👍 
Screenshot 2024-09-08 at 9 11 58 PM

**Pull request merged**